File: plugins-local/cordova-plugin-receive-content/src/android/ReceiveContent.java
package receiveContent;
import android.content.Intent;
import org.apache.cordova.CordovaPlugin;
import org.apache.cordova.CallbackContext;
import org.apache.cordova.PluginResult;
import org.json.JSONArray;
import org.json.JSONException;

/**
 * This class echoes a string called from JavaScript.
 */
public class ReceiveContent extends CordovaPlugin {

    @Override
    public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
        if (action.equals("receiveText")) {
            this.getTextFromIntent(callbackContext);
            return true;
        }
        return false;
    }

    private void getTextFromIntent (final CallbackContext context) {
        String receiveText = "";
        Intent intent = cordova.getActivity().getIntent();
        String action = intent.getAction();
        String type = intent.getType();
        if (Intent.ACTION_SEND.equals(action) && type != null) {
            if ("text/plain".equals(type)) {
                String sharedText = intent.getStringExtra(Intent.EXTRA_TEXT);
                if (sharedText != null) {
                    receiveText = sharedText;
                }
            }
        }
        context.sendPluginResult(new PluginResult(PluginResult.Status.OK, receiveText));
    }
}

File: plugins-local/cordova-plugin-receive-content/src/android/ReceiveContent.kt
package receiveContent

import android.content.Intent
import org.apache.cordova.CallbackContext
import org.apache.cordova.CordovaPlugin
import org.apache.cordova.PluginResult
import org.json.JSONArray
import org.json.JSONException

/**
 * This class echoes a string called from JavaScript.
 */
class ReceiveContent : CordovaPlugin() {

    @Throws(JSONException::class)
    override fun execute(action: String, args: JSONArray, callbackContext: CallbackContext): Boolean {
        if (action == "receiveText") {
            this.getTextFromIntent(callbackContext)
            return true
        }
        return false
    }

    private fun getTextFromIntent(context: CallbackContext) {
        var receiveText = ""
        val intent = cordova.activity.intent
        val action = intent.action
        val type = intent.type
        if (Intent.ACTION_SEND == action && type != null) {
            if (type == "text/plain") {
                val sharedText = intent.getStringExtra(Intent.EXTRA_TEXT)
                if (sharedText != null) {
                    receiveText = sharedText
                }
            }
        }
        context.sendPluginResult(PluginResult(PluginResult.Status.OK, receiveText))
    }
}